File: sample-vue-app-plugin/ios/Plugin/DataExchangerPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(DataExchangerPlugin)
public class DataExchangerPlugin: CAPPlugin {
    private let implementation = DataExchanger()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}

File: sample-vue-app-plugin/android/src/main/java/com/mycompany/plugins/example/DataExchangerPlugin.kt
package com.mycompany.plugins.example

import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import com.getcapacitor.annotation.CapacitorPlugin
import com.mycompany.plugins.example.DataExchanger
import com.getcapacitor.PluginMethod
import com.getcapacitor.PluginCall

@CapacitorPlugin(name = "DataExchanger")
class DataExchangerPlugin : Plugin() {
    private val implementation = DataExchanger()
    @PluginMethod
    fun echo(call: PluginCall) {
        val value = call.getString("value")
        val ret = JSObject()
        ret.put("value", implementation.echo(value))
        call.resolve(ret)
    }
}